Fuzzlyn/Methods/FuncBodyGenerator.cs

```diff
@@ -52,21 +55,24 @@ public void Generate(List<ScopeValue> initialScope)
 
     private StatementSyntax GenStatement(bool allowReturn = true)
     {
-        if (_finallyCount > 0)
+        if (_finallyCount > 0 || _catchCount > 0)
```

Comment:
Returns are allowed within catches, so don't think we need this.

Reply:
Why are returns not allowed within `finally`?